### PR TITLE
Feature widget dns

### PR DESCRIPTION
**Pull request by mclacore**
Adding DNS widget support to Azure AKS bundle. Things to note:
* The configuration is in such a way to easily support multiple DNS zones down the road without needing to revert a bunch of code in the process
* Currently, [only 1 DNS zone is supported with External DNS](https://stacksimplify.com/azure-aks/azure-kubernetes-service-externaldns/)
* This is a temporary setup until workload identity can be implemented

resolves: https://github.com/massdriver-cloud/massdriver/issues/1462

### CODE REVIEW

File: custom-resources/cert-manager-issuer.tf [hunk 1]
```diff
@@ -1,7 +1,11 @@
 locals {
-  dns_zones           = try(var.core_services.azure_dns_zones.dns_zones, [])
-  dns_resource_group  = try(var.core_services.azure_dns_zones.resource_group, "")
-  enable_cert_manager = length(local.dns_zones) > 0 && length(local.dns_resource_group) > 0
+  enable_azure_dns = length(var.core_services.azure_dns_zones) > 0
```
Comment (WillBeebe) [inline custom-resources/cert-manager-issuer.tf:2]: `var.core_services.azure_dns_zones` is a map not an array so `length` here is not what we're looking for. I believe you want `length(var.core_services.azure_dns_zones.dns_zone) > 0`

File: custom-resources/cert-manager-issuer.tf [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+  zone_split_ids   = local.enable_azure_dns ? [for zone in var.core_services.azure_dns_zones.dns_zone : split("/", zone)] : []
```
Comment (WillBeebe) [inline custom-resources/cert-manager-issuer.tf:3]: I'm not seeing `zone_split_ids` used anywhere else.

File: custom-resources/cert-manager-issuer.tf [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+  azure_dns_zones = local.enable_azure_dns ? { # This is hardcoded to expect a single element. Will need to change this when multiple DNS zones are supported by external DNS.
+    dns_zones      = toset([element(split("/", var.core_services.azure_dns_zones.dns_zone[0]), index(split("/", var.core_services.azure_dns_zones.dns_zone[0]), "dnszones") + 1)])
```
Comment (WillBeebe) [inline custom-resources/cert-manager-issuer.tf:5]: You could make this more readable by adding a few more locals.

```
first_zone = var.core_services.azure_dns_zones.dns_zone[0]
split_zone = split("/", local.first_zone)
dns_zones = [element(local.split_zone, index(local.split_zone, "dnszones") + 1)]
resource_group =  element(local.split_zone, index(local.split_zone, "resourceGroups") + 1)
```

File: custom-resources/cert-manager-issuer.tf [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
+    resource_group = element(split("/", var.core_services.azure_dns_zones.dns_zone[0]), index(split("/", var.core_services.azure_dns_zones.dns_zone[0]), "resourceGroups") + 1)
```
Comment (WillBeebe) [inline custom-resources/cert-manager-issuer.tf:6]: should this be `resourceGroup`? Is it plural in the string?

File: custom-resources/cert-manager-issuer.tf [hunk 5]
```diff
@@ -28,7 +32,7 @@ resource "kubernetes_manifest" "cluster_issuer" {
         "privateKeySecretRef" = {
           "name" : "letsencrypt-prod-issuer-account-key"
         },
-        "solvers" = concat([for zone in var.core_services.azure_dns_zones.dns_zones : {
+        "solvers" = concat([for zone in var.core_services.azure_dns_zones.dns_zone : {
```
Comment (WillBeebe) [inline custom-resources/cert-manager-issuer.tf:35]: we should double-check that the `solvers` expect the whole zone id instead of just the zone name.

**Review by WillBeebe — COMMENTED** [review]
Looking good, this stuff is tricky! Just a few questions / comments.

File: custom-resources/cert-manager-issuer.tf (hunk 1 above)
Comment (mclacore) [reply custom-resources/cert-manager-issuer.tf:2]: Agreed, I'll fix it.

File: custom-resources/cert-manager-issuer.tf (hunk 2 above)
Comment (mclacore) [reply custom-resources/cert-manager-issuer.tf:3]: Ah yea, we were using it by had to set something else up.

File: custom-resources/cert-manager-issuer.tf (hunk 4 above)
Comment (mclacore) [reply custom-resources/cert-manager-issuer.tf:6]: It's plural in the string:
```
➜  core-services git:(feature-widget-dns) ✗ tf console
> var.core_services.azure_dns_zones.dns_zone
[
  "/subscriptions/xxx-xxx-xxx-xxx/resourceGroups/xxx-xxx/providers/Microsoft.Network/dnszones/xxx.com",
]
>
```

File: custom-resources/cert-manager-issuer.tf (hunk 3 above)
Comment (mclacore) [reply custom-resources/cert-manager-issuer.tf:5]: Agreed, I'll fix it up.

File: custom-resources/cert-manager-issuer.tf (hunk 5 above)
Comment (mclacore) [reply custom-resources/cert-manager-issuer.tf]: Yea it expects the string. Currently testing a fix for this.

**Review by WillBeebe — APPROVED** [review]
Thanks for working through this.

**Pull request merged**